File: 프로그래머스/Lv.1/하샤드 수/하샤드 수.js
function solution(x) {
  let answer = true;
  let sum =0;
  let strx = String(x);
  for(let i=0;i<strx.length;i++){
      sum += Number(strx[i]);
  }
  console.log(sum)
  if(x % sum !== 0){
      answer = false;
  }
  return answer;
}

File: 프로그래머스/Lv.1/하샤드 수/하샤드 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x) {
    bool answer = true;
    int total =0;
    int y = x;
    while(1){
        total += y % 10;
        y = y / 10;
        if(y == 0){
            break;
        }
    }
    if(x%total != 0){
        answer = false;
    }
    return answer;
}